### ISSUE TITLE
Ubereats New Ads

### ISSUE THREAD

**Issue by aldobarr** [uBlockOrigin/uAssets#23707]
### Prerequisites

- [X] This is **NOT** a [_YouTube_](https://www.github.com/uBlockOrigin/uAssets/issues/20586), [_Facebook_](https://www.github.com/uBlockOrigin/uAssets/issues/3367), [_Twitch_](https://www.github.com/uBlockOrigin/uAssets/issues/5184) or [_a shortener/hosting site_](https://github.com/uBlockOrigin/uAssets/discussions/17361) report. _These sites MUST be reported by clicking their **respective links**_.
- [X] I read and understand the [policy about what is a valid filter issue](https://github.com/uBlockOrigin/uAssets/blob/master/README.md#uassets).
- [X] I verified that this issue is not a duplicate. (Search [here](https://github.com/uBlockOrigin/uAssets/issues) to find out.)
- [X] I did not remove any of the [default filter lists](https://github.com/uBlockOrigin/uAssets/assets/66517106/6e8d024a-5869-47cf-b142-b2357eb6c7d0), or I have verified that the issue was not caused by removing any of the default lists.
- [X] I did not enable additional filter lists, or I have verified that the issue still occurs without enabling additional filter lists.
- [X] I do not have custom filters/rules, or I have verified that the issue still occurs without custom filters/rules.
- [X] I am not using uBlock Origin along with other content blockers.
- [X] I have verified that the web browser's built-in content blocker/tracking protection, [network wide/DNS blocking](https://en.wikipedia.org/wiki/DNS_blocking), or my VPN is not causing the issue.
- [X] I have verified that other extensions are not causing the issue.
- [X] If this is about a breakage or detection, I have verified that it is caused by uBlock Origin and isn't a site issue.
- [ ] I did not answer truthfully to **ALL** the above checkboxes.

### URL(s) where the issue occurs.

```yaml
https://www.ubereats.com/orders/{ORDER-ID}
```

### Description

There are some new ads on the order page after an order has been placed.

### Other extensions used

betterTTV
bitwarden
clickbait remover for youtube
don't fuck with paste
greasemonkey (only 1 script which only interacts on twitch.tv)
hide private mode
return youtube dislike
sponsorblock for youtube
ttv lol pro
translate web pages
youtube high definition
youtube nonstop
youtube spam remover

### Screenshot(s)

<details>
<summary>Screenshot(s)</summary>

![image](https://github.com/uBlockOrigin/uAssets/assets/1071197/5aaeca6a-725f-4574-8121-cbb8be5c5ca9)

</details>

### Configuration

uBlock Origin: 1.57.2
Firefox: 125
filterset (summary):
 network: 187584
 cosmetic: 49816
 scriptlet: 24739
 html: 2061
listset (total-discarded, last-updated):
 added:
  https://gitlab.com/magnolia1234/bypass-paywalls-clean-filters/-/raw/main/bpc-paywall-filter.txt: 0-0, never
  https://raw.githubusercontent.com/liamengland1/miscfilters/master/antipaywall.txt: 2098-0, 5m
  adguard-spyware-url: 1512-120, 8m
  fanboy-thirdparty_social: 68-0, 8m
  ublock-annoyances: 5823-38, 8m
  adguard-spyware: 83498-31917, 8m
 default:
  user-filters: 5-0, never
  ublock-filters: 38196-118, 9m
  ublock-badware: 8548-0, 9m
  ublock-privacy: 907-3, 9m
  ublock-unbreak: 2332-2, 8m Δ
  ublock-quick-fixes: 193-5, 8m Δ
  easylist: 87514-169, 8m
  easyprivacy: 50907-75, 8m
  urlhaus-1: 12635-3, 8m
  plowe-0: 3729-1160, 8m
filterset (user): [array of 5 redacted]
trustedset:
 added: [array of 7 redacted]
hostRuleset:
 added: [array of 8 redacted]
urlRuleset:
 added: [array of 442 redacted]
userSettings:
 advancedUserEnabled: true
hiddenSettings: [none]
supportStats:
 allReadyAfter: 154 ms (selfie)
 maxAssetCacheWait: 43 ms
 cacheBackend: indexedDB
popupPanel:
 blocked: 74
 network:
  ubereats.com: 74
 extended:
  ##.dw.ec
  ##+js(set-constant, Navigator.prototype.globalPrivacyControl, fa…
  ##+js(set-constant, navigator.globalPrivacyControl, false)

**Comment by aldobarr** [comment]
@stephenhawk8054 I order quite often. Happy to help with my account if needed.

**Comment by stephenhawk8054** [comment]
@aldobarr The label is just to inform any others who want to look at the case that the issue needs an account to reproduce.

Can you right click on the ads -> Click on `Inspect` -> It will open Firefox devtools -> Screenshot it here? (It would be better if you can drag the devtools to be bigger before screenshot)

**Comment by aldobarr** [comment]
> @aldobarr The label is just to inform any others who want to look at the case that the issue needs an account to reproduce.
> 
> Can you right click on the ads -> Click on `Inspect` -> It will open Firefox devtools -> Screenshot it here? (It would be better if you can drag the devtools to be bigger before screenshot)

The last time I ordered I didn't get an ad. Just a self promo to upgrade ubear eats pro. Example:

![no_ad](https://github.com/uBlockOrigin/uAssets/assets/1071197/8c1d8cac-aa27-4555-9ed2-d5432fbcec50)
[no_ad.txt](https://github.com/uBlockOrigin/uAssets/files/15403409/no_ad.txt)

Thought you might want to see that too. I got the ads back this morning:
![yes_ad](https://github.com/uBlockOrigin/uAssets/assets/1071197/cc626660-d86a-4da7-948f-034a513e24b9)
[yes_ad.txt](https://github.com/uBlockOrigin/uAssets/files/15403417/yes_ad.txt)

Included the actual dom contents as a text file as well in both cases.

**Comment by aldobarr** [comment]
> @aldobarr The label is just to inform any others who want to look at the case that the issue needs an account to reproduce.
> 
> Can you right click on the ads -> Click on `Inspect` -> It will open Firefox devtools -> Screenshot it here? (It would be better if you can drag the devtools to be bigger before screenshot)

@stephenhawk8054 The last time I ordered I didn't get an ad. Just a self promo to upgrade ubear eats pro. Example:

![no_ad](https://github.com/uBlockOrigin/uAssets/assets/1071197/8c1d8cac-aa27-4555-9ed2-d5432fbcec50)
[no_ad.txt](https://github.com/uBlockOrigin/uAssets/files/15403409/no_ad.txt)

Thought you might want to see that too. I got the ads back this morning:
![yes_ad](https://github.com/uBlockOrigin/uAssets/assets/1071197/cc626660-d86a-4da7-948f-034a513e24b9)
[yes_ad.txt](https://github.com/uBlockOrigin/uAssets/files/15403417/yes_ad.txt)

Included the actual dom contents as a text file as well in both cases.

**Comment by stephenhawk8054** [comment]
Yeah, this is difficult to make if just based on the DOM. You can try this as a custom filter

```adb
ubereats.com##:matches-path(/orders) main#main-content ol[style^="transform"]:has(> li > a[href]):has-text(Ad)
```

but I'm not sure if it would cause breakages somewhere or not.

**Comment by aldobarr** [comment]
Order just got delivered so I'll try it out next time. Also realistically the only thing that matters on the order page is the delivery estimate and the map. The rest is just fluff imo.

**Comment by stephenhawk8054** [comment]
I'll convert this to a discussion. It's still a risky filter so anyone can test themselves as a custom filter
